File: No.2490-Bees-master/api/src/main/java/org/bees/biz/service/LiveService.java
package org.bees.biz.service;

import java.lang.reflect.InvocationTargetException;
import java.util.Date;
import java.util.LinkedList;
import java.util.List;

import org.apache.commons.beanutils.PropertyUtils;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.bees.biz.persistence.manager.LiveGiftManager;
import org.bees.biz.persistence.manager.LiveManager;
import org.bees.biz.persistence.manager.LivePeopleManager;
import org.bees.biz.persistence.manager.MarketManager;
import org.bees.biz.persistence.manager.UserManager;
import org.bees.biz.persistence.model.Live;
import org.bees.biz.persistence.model.LiveGift;
import org.bees.biz.persistence.model.LivePeople;
import org.bees.biz.persistence.model.Market;
import org.bees.biz.persistence.model.User;
import org.bees.biz.service.dto.LiveDto;
import org.bees.biz.service.dto.LiveGiftDto;
import org.bees.biz.service.dto.LivePeopleDto;
import org.bees.biz.service.dto.MarketDto;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import projects.commons.utils.ValidateUtils;

@Service
public class LiveService {
	
	@Autowired
	private LiveManager liveManager;
	
	@Autowired
	private LivePeopleManager livePeopleManager;
	
	@Autowired
	private LiveGiftManager liveGiftManager;
	
	@Autowired
	private MarketManager marketManager;
	
	@Autowired
	private UserManager userManager;
	
	
	@SuppressWarnings("unused")
	private final static Log LOG = LogFactory.getLog(LiveService.class);


	public List<MarketDto> market(Integer uid, Integer id, String name,
			Date startDate, Date endDate, Integer startTime, Integer endTime,
			Integer start, Integer limit) {
		List<Integer> uids=new LinkedList<Integer>();
		if(!ValidateUtils.isNull(name)||!ValidateUtils.isNull(uid)){
			List<User> users=this.userManager.adminList(null,null, name, null, null, null, null,null, uid, null,null, null, null);
			if(!ValidateUtils.isNull(users)){
				for (User user : users) {
					uids.add(user.getId());
				}
			}else{
				uids.add(-1);
			}			
		}
		List<Market> markets=this.marketManager.lists(uids,id,startDate,endDate,startTime,endTime,start,limit);
		List<MarketDto> dtos=new LinkedList<MarketDto>();
		for (Market market : markets) {
			MarketDto dto=new MarketDto();
			try {				
				PropertyUtils.copyProperties(dto, market);
				User user=this.userManager.getInfoById(market.getUid());
				dto.setName(user.getTruename());
				
			} catch (IllegalAccessException | InvocationTargetException
					| NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}
		return dtos;
	}


	public List<LivePeopleDto> people(Integer uid, Integer id, String name,
			Date startDate, Date endDate, Integer start, Integer limit) {
		List<Integer> uids=new LinkedList<Integer>();
		if(!ValidateUtils.isNull(name)||!ValidateUtils.isNull(uid)){
			List<User> users=this.userManager.adminList(null,null, name, null, null, null, null,null, uid, null, null,null, null);
			if(!ValidateUtils.isNull(users)){
				for (User user : users) {
					uids.add(user.getId());
				}
			}else{
				uids.add(-1);
			}			
		}
		List<LivePeople> peoples=this.livePeopleManager.lists(uids,id,startDate,endDate,start,limit);
		List<LivePeopleDto> dtos=new LinkedList<LivePeopleDto>();
		for (LivePeople people : peoples) {
			LivePeopleDto dto=new LivePeopleDto();
			try {				
				PropertyUtils.copyProperties(dto, people);
				User user=this.userManager.getInfoById(people.getUid());
				dto.setName(user.getTruename());				
			} catch (IllegalAccessException | InvocationTargetException
					| NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}
		return dtos;
	}


	public List<LiveDto> live(Integer uid, Integer id, String name,
			Date startDate, Date endDate, Integer startTime, Integer endTime,
			Integer start, Integer limit) {
		List<Integer> uids=new LinkedList<Integer>();
		if(!ValidateUtils.isNull(name)||!ValidateUtils.isNull(uid)){
			List<User> users=this.userManager.adminList(null,null, name, null, null, null, null,null, uid, null,null, null, null);
			if(!ValidateUtils.isNull(users)){
				for (User user : users) {
					uids.add(user.getId());
				}
			}else{
				uids.add(-1);
			}			
		}
		
		List<Live> lives=this.liveManager.lists(uids,id,startDate,endDate,startTime,endTime,start,limit);
		List<LiveDto> dtos=new LinkedList<LiveDto>();
		for (Live live : lives) {
			LiveDto dto=new LiveDto();
			try {				
				PropertyUtils.copyProperties(dto, live);
				User user=this.userManager.getInfoById(live.getUid());
				dto.setName(user.getTruename());
				
			} catch (IllegalAccessException | InvocationTargetException
					| NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}
		return dtos;
	}


	public int marketCount(Integer uid, Integer id, String name,
			Date startDate, Date endDate, Integer startTime, Integer endTime) {
		List<Integer> uids=new LinkedList<Integer>();
		if(!ValidateUtils.isNull(name)||!ValidateUtils.isNull(uid)){
			List<User> users=this.userManager.adminList(null,null, name, null, null, null, null,null, uid, null,null, null, null);
			if(!ValidateUtils.isNull(users)){
				for (User user : users) {
					uids.add(user.getId());
				}
			}else{
				uids.add(-1);
			}			
		}
		return this.marketManager.listsCount(uids,id,startDate,endDate,startTime,endTime);
	}


	public int liveCount(Integer uid, Integer id, String name, Date startDate,
			Date endDate, Integer startTime, Integer endTime) {
		List<Integer> uids=new LinkedList<Integer>();
		if(!ValidateUtils.isNull(name)||!ValidateUtils.isNull(uid)){
			List<User> users=this.userManager.adminList(null,null, name, null, null, null, null,null, uid, null, null,null, null);
			if(!ValidateUtils.isNull(users)){
				for (User user : users) {
					uids.add(user.getId());
				}
			}else{
				uids.add(-1);
			}			
		}
		return this.liveManager.listsCount(uids,id,startDate,endDate,startTime,endTime);
	}


	public int peopleCount(Integer uid, Integer id, String name,
			Date startDate, Date endDate) {
		List<Integer> uids=new LinkedList<Integer>();
		if(!ValidateUtils.isNull(name)||!ValidateUtils.isNull(uid)){
			List<User> users=this.userManager.adminList(null,null, name, null, null, null, null,null, uid, null, null,null, null);
			if(!ValidateUtils.isNull(users)){
				for (User user : users) {
					uids.add(user.getId());
				}
			}else{
				uids.add(-1);
			}			
		}
		return this.livePeopleManager.listsCount(uids,id,startDate,endDate);
	}


	public List<LiveGiftDto> gift(Integer id, Date startDate, Date endDate,
			Integer startTime, Integer endTime, Integer start, Integer limit) {
		
		List<LiveGift> gifts=this.liveGiftManager.lists(id,startDate,endDate,startTime,endTime,start,limit);
		List<LiveGiftDto> dtos=new LinkedList<LiveGiftDto>();
		for (LiveGift gift : gifts) {
			LiveGiftDto dto=new LiveGiftDto();
			try {				
				PropertyUtils.copyProperties(dto, gift);
				Live live=this.liveManager.getInfoById(gift.getLiveId());
				User user=this.userManager.getInfoById(live.getUid());
				dto.setUsername(user.getTruename());
				dto.setUid(user.getId());
			} catch (IllegalAccessException | InvocationTargetException
					| NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}
		return dtos;
	}


	public int giftCount(Integer id, Date startDate, Date endDate,
			Integer startTime, Integer endTime) {
		return this.liveGiftManager.listsCount(id,startDate,endDate,startTime,endTime);
	}


	
}

File: beesInterface/bees_live/liveService.py
# coding=utf-8
from bees_user.models import User
from exception.serviceException import ServiceException
from utils.validateUtils import ValidateUtils
from models import Market, LivePeople, Live, LiveGift
from utils.const import Const

class LiveService:
	@staticmethod
	def market(uid,id,name,startDate,endDate,startTime,endTime,start,limit):
		uids=[]
		if not ValidateUtils.isNull(name) or (not ValidateUtils.isNull(uid)):
			users = User.objects.all()
			if not ValidateUtils.isNull(name):
				users = users.filter(truename__contains=name)
			if not ValidateUtils.isNull(uid):
				users = users.filter(id=uid)
			if not ValidateUtils.isNull(users):
				for user in users:
					uids.append(user.id)
			else:
				uids.append(-1)
		markets = Market.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(uids):
			markets = markets.filter(uid__in=uids)
		if not ValidateUtils.isNull(id):
			markets = markets.filter(id=id)
		if not ValidateUtils.isNull(startDate):
			markets = markets.filter(dates__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			markets = markets.filter(dates__lte=endDate)
		if not ValidateUtils.isNull(startTime):
			markets = markets.filter(times__gte=startTime)
		if not ValidateUtils.isNull(endTime):
			markets = markets.filter(times__lte=endTime)
		if ValidateUtils.isNull(start):
			start = 0
		else:
			start = int(start)
		if ValidateUtils.isNull(limit):
			markets = markets[start:]
		else:
			limit = int(limit)
			markets = markets[start: start+limit]
		dtos=[]
		for market in markets:
			dtos.append(market.getMarketDto())
		return dtos

	@staticmethod
	def people(id,uid,name,startDate,endDate,startTime,endTime,start,limit):
		uids=[]
		if not ValidateUtils.isNull(name) or not ValidateUtils.isNull(uid):
			users = User.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
			if not ValidateUtils.isNull(name):
				users = users.filter(truename__contains=name)
			if not ValidateUtils.isNull(uid):
				users = users.filter(id=uid)
			if not ValidateUtils.isNull(users):
				for user in users:
					uids.append(user.id)
			else:
				uids.append(-1)
		peoples = LivePeople.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(uids):
			peoples = peoples.filter(uid__in=uids)
		if not ValidateUtils.isNull(id):
			peoples = peoples.filter(id=id)
		if not ValidateUtils.isNull(startDate):
			peoples = peoples.filter(dates__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			peoples = peoples.filter(dates__lte=endDate)
		if ValidateUtils.isNull(start):
			start = 0
		else:
			start = int(start)
		if ValidateUtils.isNull(limit):
			peoples = peoples[start:]
		else:
			limit = int(limit)
			peoples = peoples[start: start+limit]
		dtos=[]
		for people in peoples:
			dtos.append(people.getLivePeopleDto())
		return dtos

	@staticmethod
	def live(id,uid,name,startDate,endDate,startTime,endTime,start,limit):
		uids=[]
		if not ValidateUtils.isNull(name) or not ValidateUtils.isNull(uid):
			users = User.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
			if not ValidateUtils.isNull(name):
				users = users.filter(truename__contains=name)
			if not ValidateUtils.isNull(uid):
				users = users.filter(id=uid)
			if not ValidateUtils.isNull(users):
				for user in users:
					uids.append(user.id)
			else:
				uids.append(-1)
		lives = Live.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(id):
			lives = lives.filter(id=id)
		if not ValidateUtils.isNull(uids):
			lives = lives.filter(uid__in=uids)
		if not ValidateUtils.isNull(startDate):
			lives = lives.filter(dates__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			lives = lives.filter(dates__lte=endDate)
		if not ValidateUtils.isNull(startTime):
			lives = lives.filter(times__gte=startTime)
		if not ValidateUtils.isNull(endTime):
			lives = lives.filter(times__lte=endTime)
		if ValidateUtils.isNull(start):
			start = 0
		else:
			start = int(start)
		if ValidateUtils.isNull(limit):
			lives = lives[start:]
		else:
			limit = int(limit)
			lives = lives[start: start+limit]
		dtos=[]
		for live in lives:
			dtos.append(live.getLiveDict())
		return dtos

	@staticmethod
	def marketCount(id,uid,name,startDate,endDate,startTime,endTime):
		uids=[]
		if not ValidateUtils.isNull(name) or (not ValidateUtils.isNull(uid)):
			users = User.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
			if not ValidateUtils.isNull(name):
				users = users.filter(truename__contains=name)
			if not ValidateUtils.isNull(uid):
				users = users.filter(id=uid)
			if not ValidateUtils.isNull(users):
				for user in users:
					uids.append(user.id)
			else:
				uids.append(-1)
		markets = Market.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL)
		if not ValidateUtils.isNull(uids):
			markets = markets.filter(uid__in=uids)
		if not ValidateUtils.isNull(id):
			markets = markets.filter(id=id)
		if not ValidateUtils.isNull(startDate):
			markets = markets.filter(dates__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			markets = markets.filter(dates__lte=endDate)
		if not ValidateUtils.isNull(startTime):
			markets = markets.filter(times__gte=startTime)
		if not ValidateUtils.isNull(endTime):
			markets = markets.filter(times__lte=endTime)
		return markets.count()

	@staticmethod
	def liveCount(id,uid,name,startDate,endDate,startTime,endTime):
		uids=[]
		if not ValidateUtils.isNull(name) or (not ValidateUtils.isNull(uid)):
			users = User.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
			if not ValidateUtils.isNull(name):
				users = users.filter(truename__contains=name)
			if not ValidateUtils.isNull(uid):
				users = users.filter(id=uid)
			if not ValidateUtils.isNull(users):
				for user in users:
					uids.append(user.id)
			else:
				uids.append(-1)
		peoples = LivePeople.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL)
		if not ValidateUtils.isNull(uids):
			peoples = peoples.filter(uid__in=uids)
		if not ValidateUtils.isNull(id):
			peoples = peoples.filter(id=id)
		if not ValidateUtils.isNull(startDate):
			peoples = peoples.filter(dates__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			peoples = peoples.filter(dates__lte=endDate)
		if not ValidateUtils.isNull(startTime):
			peoples = peoples.filter(times__gte=startTime)
		if not ValidateUtils.isNull(endTime):
			peoples = peoples.filter(times__lte=endTime)
		return peoples.count()

	@staticmethod
	def gift(id, startDate, endDate, startTime, endTime, start, limit):
		gifts = LiveGift.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(id):
			gifts = gifts.filter(id=id)
		if not ValidateUtils.isNull(startDate):
			gifts = gifts.filter(dates__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			gifts = gifts.filter(dates__lte=endDate)
		if not ValidateUtils.isNull(startTime):
			gifts = gifts.filter(times__gte=startTime)
		if not ValidateUtils.isNull(endTime):
			gifts = gifts.filter(times__lte=endTime)
		if ValidateUtils.isNull(start):
			start = 0
		else:
			start = int(start)
		if ValidateUtils.isNull(limit):
			gifts = gifts[start:]
		else:
			limit = int(limit)
			gifts = gifts[start: start+limit]
		dtos=[]
		for gift in gifts:
			dtos.append(gift.getLiveGiftDto())
		return dtos

	@staticmethod
	def giftCount(id, startDate, endDate, startTime, endTime):
		gifts = LiveGift.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL)
		if not ValidateUtils.isNull(id):
			gifts = gifts.filter(id=id)
		if not ValidateUtils.isNull(startDate):
			gifts = gifts.filter(dates__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			gifts = gifts.filter(dates__lte=endDate)
		if not ValidateUtils.isNull(startTime):
			gifts = gifts.filter(times__gte=startTime)
		if not ValidateUtils.isNull(endTime):
			gifts = gifts.filter(times__lte=endTime)
		return gifts.count()